compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPU2DScalableTo1DScalable.cpp

```diff
@@ -0,0 +1,193 @@
+    bool isArmSME = assumeArmSME || hasSMEFeature(targetAttr);
+    if (!isArmSME || opKnownToSupport2DScalableVectorizationWithArmSME(op))
+      return failure();
+
+    auto loweringConfigAttr = getLoweringConfig(op);
+    if (!loweringConfigAttr)
+      return failure();
+
+    auto tileSizes = loweringConfigAttr.getTileSizeVals();
+    auto scalableFlags = loweringConfigAttr.getScalableTileFlagVals();
+
+    // 1. Drop scalable dimensions from leading tiling levels first (and leading
+    // dimensions within tiling levels). This works out as dropping scalability
+    // from leading dimensions of the vector type.
+    int64_t firstTilingLevelWithScalableDims = -1;
+    int64_t numScalableDims = 0;
+    for (auto [level, levelScalableFlags] : llvm::enumerate(scalableFlags)) {
+      numScalableDims += llvm::count(levelScalableFlags, true);
+      if (numScalableDims > 0 && firstTilingLevelWithScalableDims == -1)
+        firstTilingLevelWithScalableDims = level;
+    }
+
+    if (numScalableDims <= 1)
+      return failure();
+
+    auto levelTileSizes = tileSizes[firstTilingLevelWithScalableDims];
+    auto levelScalableFlags = scalableFlags[firstTilingLevelWithScalableDims];
+
+    SmallVector<int64_t> loopTileSizes;
+    SmallVector<bool> newScalableFlags;
+    for (auto [flag, size] : llvm::zip(levelScalableFlags, levelTileSizes)) {
+      if (flag && numScalableDims >= 2) {
+        --numScalableDims;
+        loopTileSizes.push_back(size);
+        newScalableFlags.push_back(false);
+      } else {
+        loopTileSizes.push_back(0);
+        newScalableFlags.push_back(flag);
+      }
+    }
+
+    // 2. Re-tile the operation with some scalability dropped. This introduces
+    // loops for previously scalable vector/tile sizes.
+    scf::SCFTilingOptions options{};
+    setSCFTileSizes(options, op, loopTileSizes, {});
+    auto tilingResult =
+        scf::tileUsingSCF(rewriter, cast<TilingInterface>(op), options);
+    if (failed(tilingResult))
+      return failure();
+
+    // 3. Update the lowering config of the new tiled operations.
+    scalableFlags[firstTilingLevelWithScalableDims] = newScalableFlags;
+    auto newLoweringConfig = IREE::Codegen::LoweringConfigAttr::get(
+        getContext(), tileSizes, scalableFlags);
+    for (auto *newOp : tilingResult->tiledOps) {
+      if (isa<TilingInterface>(newOp))
+        setLoweringConfig(newOp, newLoweringConfig);
+    }
+
+    rewriter.replaceOp(op, tilingResult->replacements);
+    return success();
+  };
+
+private:
+  bool assumeArmSME{false};
+};
+
+void LLVMCPU2DScalableTo1DScalablePass::runOnOperation() {
+  RewritePatternSet patterns(&getContext());
+  patterns.add<DropUnsupportedScalableDimsFromTilingInterfaceOps>(
+      patterns.getContext(), assumeArmSME);
+
+  if (mlir::failed(mlir::applyPatternsAndFoldGreedily(getOperation(),
+                                                      std::move(patterns)))) {
```

Comment:
errant extra line?

Reply:
This is just the result of `clang-format` line wrapping :pensive:

Reply:
One thing you can do is drop the `mlir::` since you're already in the `mlir::iree_compiler` namespace:
```c++
  if (failed(
          applyPatternsAndFoldGreedily(getOperation(), std::move(patterns)))) {
    signalPassFailure();
  }
```

I don't actually see any uses of `mlir::applyPatternsAndFoldGreedily` or `mlir::failed(` in IREE, all omit that prefix.